definitions/procedures/foreman/remove_duplicate_permissions.rb

```diff
@@ -0,0 +1,80 @@
+module Procedures::Foreman
+  class RemoveDuplicatePermissions < ForemanMaintain::Procedure
+    metadata do
+      for_feature :foreman_database
+      description 'Remove duplicate permissions from DB'
+    end
+
+    # rubocop:disable Metrics/MethodLength
+    def run
+      duplicate_permissions = feature(:foreman_database).query(
+        Checks::Foreman::CheckDuplicatePermission.query_to_get_duplicate_permission
+      )
+      duplicate_permissions.each do |permission|
+        assigned_permissions = []
+        ids = feature(:foreman_database).query(query_to_get_permission_ids(permission)).
+              flat_map(&:values)
+        ids.each do |permission_id|
+          filterings = feature(:foreman_database).
+                       query(query_to_check_permission_assign_to_filter(permission_id))
+          if filterings.empty?
+            delete_permission(permission_id)
+          else
+            assigned_permissions << permission_id
+          end
+        end
+        if assigned_permissions.length > 1
+          fix_permissions(assigned_permissions)
+        end
+      end
+    end
+    # rubocop:enable Metrics/MethodLength
+
+    private
+
+    def fix_permissions(assigned_permissions)
+      persist_permission = assigned_permissions.shift
+      assigned_permissions.each do |permission|
+        update_filtering(permission, persist_permission)
+        delete_permission(permission)
+      end
+    end
+
+    def update_filtering(old_id, new_id)
+      sql = <<-SQL
+      WITH rows AS (
+        UPDATE filterings SET permission_id = '#{new_id}' WHERE permission_id = '#{old_id}'
```

Comment:
As we are keeping one permission_id and replacing all other records with same, this is also eligible for `in` clause ?